templates/etc/init.d/rpm/td-agent

```diff
@@ -148,7 +148,9 @@ if [ -n "${TD_AGENT_USER}" ]; then
   fi
   mkdir -p "$(dirname "${TD_AGENT_PID_FILE}")"
   chown -R "${TD_AGENT_USER}" "$(dirname "${TD_AGENT_PID_FILE}")"
-  START_STOP_DAEMON_ARGS="${START_STOP_DAEMON_ARGS} --user ${TD_AGENT_USER}"
+  # Work around https://github.com/fluent/fluentd/issues/2735
+  # TD_AGENT_ARGS="${TD_AGENT_ARGS} --daemon ${TD_AGENT_PID_FILE}"
+  START_STOP_DAEMON_ARGS="${START_STOP_DAEMON_ARGS} --background --make-pidfile"
```

Comment:
Have we tested this on CentOS/Red Hat? I didn't think this was going to work in Fedora-based distros. Both `--background` and `--make-pidfile` are arguments to `start-stop-daemon`, which is a Debian thing. This script uses the `daemon` command from `/etc/init.d/functions`, which doesn't support those. Unfortunately, my quick search didn't find a good Red Hat-compatible alternative to those args for processes that don't support daemon mode. I _think_ `--background` can be emulated by appending `\&` to the end of `TD_AGENT_ARGS`, but I don't see anything for `--make-pidfile`. We may have to research it further.